### ISSUE TITLE
Holdout_id and prediction/forecasting

### ISSUE THREAD

**Issue by Nir4** [google/meridian#802]
Hi,
E.g. we have five weeks of data so our holdout_id will be [False, False , True, True] if we want to holdout the last two weeks? .Then we run the model again but now [False, False ,False, True] holding only the last week. so basically we did a bit of cross validation. Is this correct how I defined holdout_id for this situation?

But then its mentioned in the documentation:

"We recommend using a holdout sample that is fairly balanced across geos and time periods. In other words, use a holdout sample that has approximately the same number of holdout observations for each geo and approximately the same number of holdout observations for each time period. If the holdout sample is imbalanced, this can result in too few training observations to estimate the geo effect 𝜏𝑔
 for certain geos, or the time effect 𝜇𝑡for certain time periods."

That means cross validation is not preferred because the no. of holdout observations would vary because we want the training samples to be contiguous so we only change how many samples we holdout from the end.

Also I have read that forecasting is not preferred for Meridian. Then what is the use of hold_id? Here prediction will always be forecasting because we will always have the training data taking first contiguous from the start and test afterwards....

**Comment by cpulavarthi** [comment]
Hello @Nir4,

Thank you for contacting us!

To answer your first question, `holdout_id` in the [`ModelSpec`](https://developers.google.com/meridian/reference/api/meridian/model/spec/ModelSpec) is a boolean array of dimensions (`n_geos`, `n_times`) for a geo-level model or (`n_times`, ) for a national-level model. So the array you shared can be passed as a `holdout_id` if you are using a national-level weekly model with only 4 weeks (not 5 as you mentioned in your question). And yes, the records corresponding to the `True` value in the `holdout_id` are set aside as part of the holdout sample, and thus are excluded from the training sample. Please note that only KPI data is excluded from the training sample, and media data is still included as it can affect Adstock for subsequent weeks.

The purpose of `holdout_id` is not forecasting but to measure/evaluate out-of-sample predictive accuracy (see [here](https://developers.google.com/meridian/docs/basics/holdout-observations#:~:text=The%20primary%20use%20of%20the,uses%20the%20same%20holdout%20sample)). You can use a cross-validation strategy as long as the validation set is a fairly balanced set in that strategy. We don’t need the training samples, and consequently the holdout set to be contiguous. It is in fact the opposite, we need the holdout set to be evenly spread out in Meridian. Therefore, commonly used cross-validation strategies such as k-fold CV or stratified CV won’t be applicable unless modified to ensure the holdout sets are balanced and distributed according to Meridian's recommendations. As mentioned earlier, only the KPI data is excluded from the training process, while the independent variables' data from the holdout periods is still used in model training, so the training sample doesn't have to be contiguous.

The primary purpose of Meridian is to identify and estimate the causal relationships between the intervenable treatment variables and the target KPI. This can help understand the contributions of various media efforts towards the target KPI, identify the saturation points, optimize budget allocations, estimate the impact of the optimized budget on the target KPI, etc. In Meridian, the `holdout_id` is applicable to check whether the model has converged to the right set of posterior parameters. Since we aren’t attempting to forecast into future unobserved periods, we emphasize using a balanced holdout set across geos and time periods to ensure a comprehensive evaluation of the model's underlying causal estimates.

I hope this answers all your questions. Feel free to reach out if you have any further queries or suggestions regarding Meridian.

Google Meridian Support Team
